janusgraph-core/src/main/java/org/janusgraph/graphdb/database/idassigner/VertexIDAssigner.java

```diff
@@ -297,7 +297,7 @@ private long getPartitionID(final InternalVertex v) {
 
     private void assignID(final InternalElement element, final long partitionIDl, final IDManager.VertexIDType userVertexIDType) {
         Preconditions.checkNotNull(element);
-        Preconditions.checkArgument(!element.hasId());
+        if (element.hasId()) return;
```

Comment:
Adding some explanation:
https://github.com/JanusGraph/janusgraph/blob/8b2dd99b5dc4a9d4b29be168769aacd059d8a12e/janusgraph-core/src/main/java/org/janusgraph/graphdb/database/StandardJanusGraph.java#L674-L685

When `hasAssignIDsImmediately` is disabled, `idAssigner::assignIDs` is called, which in turn (skipping a few function calls within `VertexIDAssigner`) calls this `assignId` method.

If the given `element` is an existing element, e.g. an existing edge, then this `Preconditions.checkArgument(!element.hasId())` will fail. Thus, we should not throw an exception here. Moreover, we should `return` immediately, as we do not want to assign a new ID to it.